pkg/hubble/exporter/config.go

```diff
@@ -0,0 +1,242 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package exporter
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"reflect"
+	"slices"
+
+	"github.com/sirupsen/logrus"
+	"sigs.k8s.io/yaml"
+
+	flowpb "github.com/cilium/cilium/api/v1/flow"
+	v1 "github.com/cilium/cilium/pkg/hubble/api/v1"
+	"github.com/cilium/cilium/pkg/time"
+)
+
+var _ ExporterConfigParser = (*exporterConfigParser)(nil)
+
+type exporterConfigParser struct {
+	logger logrus.FieldLogger
+}
+
+// NewExporterConfigParser is an ExporterConfig parser for DynamicExportersConfig.
+func NewExporterConfigParser(logger logrus.FieldLogger) *exporterConfigParser {
+	return &exporterConfigParser{logger}
+}
+
+// Parse implements ExporterConfigParser.
+func (e *exporterConfigParser) Parse(r io.Reader) (map[string]ExporterConfig, error) {
+	var buf bytes.Buffer
+	if _, err := buf.ReadFrom(r); err != nil {
+		return nil, fmt.Errorf("failed to read config: %w", err)
+	}
+	var config DynamicExportersConfig
+	if err := yaml.Unmarshal(buf.Bytes(), &config); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal yaml config: %w", err)
+	}
```

Comment:
I believe you could use `yaml.NewDecoder(r)` instead of creating a buffer and calling `yaml.Unmarshal` separately.

Reply:
It's not the same yaml implementation. This one is using `sigs.k8s.io/yaml` for some reason, which can only work on byte slices. I am unsure if changing the implementation has implications so I havent tried to.

Reply:
Hmm, yeah I think it's because they serialize everything to JSON then convert to YAML so it ends up needing to buffer everything anyways.